### PR TITLE
Python 3: fix MODBUS

### PR DESCRIPTION
**Pull request by gpotter2**

### CODE REVIEW

File: scapy/contrib/modbus.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ class ModbusWriteFileSubRequest(Packet):
     fields_desc = [ByteField("refType", 0x06),
                    ShortField("fileNumber", 0x0001),
                    ShortField("recordNumber", 0x0000),
-                   BitFieldLenField("recordLength", None, 16, length_of="recordData", adjust=lambda pkt, p: p/2),
+                   BitFieldLenField("recordLength", None, 16, length_of="recordData", adjust=lambda pkt, p: p//2),
```
Comment (johnthagen) [inline scapy/contrib/modbus.py]: Doesn't this require `from __future__ import division` to be compatible with 2.7? I don't see it being imported.

Comment (gpotter2) [reply scapy/contrib/modbus.py]: No it doesn't ^^ Python 2.7 has a native support of that

Comment (johnthagen) [reply scapy/contrib/modbus.py]: Oh, right right. You need it `division` if you want `/` to be treated as float division.

Seems like a good step towards keeping things code base compatible with both would be to use `__future__` `absolute_import` `division` and `print_function` on all modules, right?

Comment (johnthagen) [reply scapy/contrib/modbus.py]: And btw, thanks for helping push Python 3 compatibility for scapy 😃

Comment (gpotter2) [reply scapy/contrib/modbus.py]: Again, in that case it's expected to be an int, not a float

File: scapy/contrib/modbus.py [hunk 2]
```diff
@@ -773,10 +773,10 @@ class ModbusADURequest(Packet):
                    XByteField("unitId", 0xff)]  # 0xFF (recommended as non-significant value) or 0x00
 
     def guess_payload_class(self, payload):
-        function_code = int(payload[0].encode("hex"), 16)
+        function_code = int(bytes_hex(chb(payload[0])), 16)
```
Comment (p-l-) [inline scapy/contrib/modbus.py]: WTF (I'm talking about the existing code)? Looks like `function_code = orb(payload[0])` is enough, or did I miss something?

Comment (gpotter2) [reply scapy/contrib/modbus.py]: You're very right

**Pull request merged**